### DISCUSSION TITLE
Why I can’t install pytgcalls?

### DISCUSSION THREAD

**Discussion by mahdizojaji** [MarshalX/tgcalls#44] [Q&A] [answered]
> pip3 install pytgcalls
```
Requirement already satisfied: pytgcalls in /home/mr/Projects/test_tgcall/.venv/lib/python3.8/site-packages/pytgcalls-0.0.9-py3.8.egg (0.0.9)
Collecting tgcalls==0.0.5 (from pytgcalls)
  Could not find a version that satisfies the requirement tgcalls==0.0.5 (from pytgcalls) (from versions: )
No matching distribution found for tgcalls==0.0.5 (from pytgcalls)
```

**Comment by MarshalX** [comment, marked as answer]
There are 3 reasons:
- Old pip (you need to upgrade to 20+)
- Invalid Python version (3.7-3.11 is supported)
- Invalid CPU architecture  (x86_64 and ARM64 is supported)

[Guide for termux (**only for ARM64**)](https://github.com/MarshalX/tgcalls/discussions/44#discussioncomment-797659)

**Comment by mahdizojaji** [comment]
I upgrade pip & it fixed.

**Reply by lankeshhhRa1** [reply]
Congrats 👏👏

**Comment by tungszyen** [comment]
**Guide for Android** 
 
Use AnLinux to get debian for termux or Debian No Root. Both the apps (AnLinux & Debian No Root) are available in Play store.

AnLinux: https://play.google.com/store/apps/details?id=exa.lnx.a
Debian noroot: https://play.google.com/store/apps/details?id=com.cuntubuntu

What I did :
```bash
sudo apt update
sudo apt upgrade
```

Now just `sudo apt-get install` the modules: _python3_, _python3-pip_ 

Update the pip with

```bash
pip3 install --upgrade pip
```

Finally install the library

```bash
pip3 install pytgcalls
```

<hr>

For Termux
```bash
python3 -m pip install pytgcalls
```

**This worked for me in Android!**

**Reply by MarshalX** [reply]
Thanks for your research!
